### PR TITLE
Adds a recipe for the gasoline lantern.

### PR DESCRIPTION
**Pull request by Rivet-the-Zombie**
Same ingredients as the oil lamp with increased difficulty and time
requirement.

### CODE REVIEW

File: data/json/recipes/recipe_others.json [hunk 1]
```diff
@@ -419,6 +419,36 @@
...
+      [ "scrap", 1 ]
+    ],
+    [
+      [ "rag", 1 ]
+    ],
+    [
+      [ "bottle_glass", 1 ],
+      [ "flask_glass", 1],
+      [ "jar_glass", 1 ],
+      [ "clay_teapot", 1 ],
```
Comment (KA101) [inline data/json/recipes/recipe_others.json]: Hmm, if it's enclosed clay pots might not be the best choice.

Comment (Rivet-the-Zombie) [inline data/json/recipes/recipe_others.json]: At some point the recipe for the oil lamp got messed with. I'll fix both recipes now.

**Comment by Rivet-the-Zombie** [comment]
Now both (oil lamp and gas lamp) recipes require a glass component and a fuel receptacle, instead of smooshing both into the same item.

File: data/json/recipes/recipe_others.json (hunk 1 above)
Comment (KA101) [inline data/json/recipes/recipe_others.json]: Yeah, it kinda worked for oil lamps (KA101 thinks Aladdin-style lamps) but not here.  Fair enough.

**Pull request merged**